ui/src/main/java/com/hyperwallet/android/ui/transfermethod/AddTransferMethodFragment.java

```diff
@@ -289,26 +283,50 @@ public void notifyTransferMethodAdded(@NonNull final HyperwalletTransferMethod t
     }
 
     @Override
-    public void showTransferMethodFields(@NonNull final List<HyperwalletField> fields) {
+    public void showTransferMethodFields(@NonNull final List<HyperwalletFieldGroup> fields) {
         mDynamicContainer.removeAllViews();
-        int previousView = 0;
+        int previousView;
+
         try {
-            for (final HyperwalletField field : fields) {
-                AbstractWidget widget = WidgetFactory
-                        .newWidget(field, this, getContext(),
-                                mWidgetInputStateHashMap.containsKey(field.getName()) ?
-                                        mWidgetInputStateHashMap.get(field.getName()).getValue() : "",
-                                mCreateTransferMethodButton);
-                if (mWidgetInputStateHashMap.isEmpty() || !mWidgetInputStateHashMap.containsKey(widget.getName())) {
-                    mWidgetInputStateHashMap.put(widget.getName(), widget.getWidgetInputState());
+            Locale locale = new Locale.Builder().setRegion(mCountry).build();
+            // group
+            for (HyperwalletFieldGroup group : fields) {
+                String sectionHeaderText;
+                if (group.getGroupName().equals(HyperwalletFieldGroup.GroupTypes.ACCOUNT_INFORMATION)) {
+                    sectionHeaderText = requireContext().getResources()
+                            .getString(R.string.account_information_section_header, locale.getDisplayName(), mCurrency);
+                } else {
+                    sectionHeaderText = requireContext().getString(requireContext().getResources()
```

Comment:
@peter-joseph  there is we have a problem: what if we don't have such resource?
I propose at least:
 final int stringIdentifier = requireContext().getResources().getIdentifier(
                            group.getGroupName().toLowerCase(Locale.ROOT), "string", requireContext().getPackageName());
                    sectionHeaderText = stringIdentifier == 0 ? group.getGroupName() : requireContext().getString(
                            requireContext().getResources()
                                    .getIdentifier(group.getGroupName().toLowerCase(Locale.ROOT), "string",
                                            requireContext().getPackageName()));

Reply:
no, I need parameters to passed to the other resource